java/vector/src/main/java/org/apache/arrow/vector/compare/RangeEqualsVisitor.java

```diff
@@ -38,37 +38,51 @@
 public class RangeEqualsVisitor implements VectorVisitor<Boolean, Range> {
   private ValueVector left;
   private ValueVector right;
-  private boolean isTypeCheckNeeded;
+
+  private TypeEqualsVisitor typeVisitor;
+  private boolean checkType;
   private boolean typeCompareResult;
 
+  /**
+   * Constructs a new instance with default type visitor (do not check name and metadata).
+   */
+  public RangeEqualsVisitor(ValueVector left, ValueVector right, boolean checkType) {
+    this (left, right, checkType, new TypeEqualsVisitor(right));
+  }
+
   /**
    * Constructs a new instance.
    *
    * @param left left vector
    * @param right right vector
-   * @param isTypeCheckNeeded type check needed
+   * @param checkType whether need check type
+   * @param typeVisitor type visitor
    */
-  public RangeEqualsVisitor(ValueVector left, ValueVector right, boolean isTypeCheckNeeded) {
+  public RangeEqualsVisitor(
+      ValueVector left,
+      ValueVector right,
+      boolean checkType,
+      TypeEqualsVisitor typeVisitor) {
     this.left = left;
     this.right = right;
-    this.isTypeCheckNeeded = isTypeCheckNeeded;
+    this.checkType = checkType;
+    this.typeVisitor = typeVisitor;
 
     Preconditions.checkArgument(left != null,
         "left vector cannot be null");
     Preconditions.checkArgument(right != null,
         "right vector cannot be null");
+    Preconditions.checkArgument(typeVisitor.getRight() == this.right);
```

Comment:
maybe it would be better to take a BiFunction<ValueVector, ValueVector, Bool> that returns true if the types of two vectors are equal and false otherwise?

Reply:
Does this mean use BiFunction instead TypeEqualsVisitor? I think they have the same effects and we already has visitor.

Reply:
Yes, BiFunction instead of TypeEqualsVisitor.  BiFunction is more generic and can consolidate the details of using TypeEqualsVisitor into a single place instead of having to assert TypeEqualsVisitor is setup correctly.

Reply:
ok, revised.